### PR TITLE
Add mocks for docker singleton

### PR DESCRIPTION
**Pull request by nre-ableton**
Although this singleton is not part of the standard Jenkins pipeline
API, it is a frequently-used plugin and the addition of mocks for this
singleton have been requested by several users.

---

Closes https://github.com/jenkinsci/JenkinsPipelineUnit/issues/242

### CODE REVIEW

**Comment by nre-ableton** [comment]
@stchar Sorry, I did not run the whole unit test suite before creating the PR. Let me fix the tests and I'll ping you when this PR is ready for review.

**Comment by nre-ableton** [comment]
@stchar Ok, I got everything working. I guess the problem was that the `DockerMock` class was not serializable. Implementing that interface didn't completely solve the problem because that class had a `PipelineTestHelper` field, which was set in the constructor from the `BasePipelineTest` class. The presence of this field meant that the class could not be correctly serialized.

I fixed this issue by removing the field, which means that closures are executed by running the body directly instead of invoking `helper.callClosure()`. I'm not exactly sure what the implications of this are, do you?

File: src/test/jenkins/jenkinsfiles/Docker_Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import com.lesfurets.jenkins.unit.DockerMock as DockerMock
```
[the remark this answers is not in the document]
Comment (stchar) [reply src/test/jenkins/jenkinsfiles/Docker_Jenkinsfile:1]: you can use just
```
def image = docker.build(testImageName)
```
so no `import` is required

[the remark this answers is not in the document]
Comment (nre-ableton) [reply src/test/jenkins/jenkinsfiles/Docker_Jenkinsfile]: @stchar we use `import` in Jenkinsfiles all the time, but I'm fine with using `def` as you suggest.

**Pull request merged**